src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs

```diff
@@ -64,60 +62,44 @@ public bool RemoveCachedSemanticModel(SyntaxTree tree)
 
             internal DeclarationAnalysisData GetOrComputeDeclarationAnalysisData(
                 SyntaxReference declaration,
-                Func<Func<DeclarationAnalysisDataBuilder>, DeclarationAnalysisDataBuilder> computeDeclarationAnalysisData,
+                Func<DeclarationAnalysisData> computeDeclarationAnalysisData,
                 bool cacheAnalysisData)
             {
-                DeclarationAnalysisDataBuilder dataBuilder = null;
-                try
+                if (!cacheAnalysisData)
                 {
-                    if (!cacheAnalysisData)
-                    {
-                        dataBuilder = computeDeclarationAnalysisData(_declarationAnalysisDataBuilderPool.Allocate);
-                        return DeclarationAnalysisData.CreateFrom(dataBuilder);
-                    }
+                    return computeDeclarationAnalysisData();
+                }
 
-                    DeclarationAnalysisData data;
-                    lock (_declarationAnalysisDataMap)
+                DeclarationAnalysisData data;
+                lock (_declarationAnalysisDataMap)
+                {
+                    if (_declarationAnalysisDataMap.TryGetValue(declaration, out data))
                     {
-                        if (_declarationAnalysisDataMap.TryGetValue(declaration, out data))
-                        {
-                            return data;
-                        }
+                        return data;
                     }
+                }
 
-                    dataBuilder = computeDeclarationAnalysisData(_declarationAnalysisDataBuilderPool.Allocate);
+                data = computeDeclarationAnalysisData();
 
-                    lock (_declarationAnalysisDataMap)
+                lock (_declarationAnalysisDataMap)
+                {
+                    if (!_declarationAnalysisDataMap.TryGetValue(declaration, out DeclarationAnalysisData existingData))
```

Comment:
out var?

Reply:
You do a tryGet just so you can know to add.  That's two lookups.  Why not just put the value in the map unilaterally?